content/client-lib-development-guide/features.textile

```diff
@@ -270,7 +272,9 @@ h3(#realtimeclient). RealtimeClient
 * @(RTC6)@ @RealtimeClient#time@ function:
 ** @(RTC6a)@ Proxy to @RestClient#time@ presented with an async or threaded interface as appropriate
 * @(RTC7)@ The client library must use the configured timeouts specified in the @ClientOptions@, falling back to the "client library defaults":#defaults and defaults described in @ClientOptions@ below
-* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth.authorise@ is called with @AuthOptions#force@ set to @true@ and the state is @CONNECTED@, @CONNECTING@, or @DISCONNECTED@, the client will obtain a new token, disconnect the current transport and resume the connection (see "@RTN15c@":#RTN15c for details on resume). A test should exist that performs an upgrade of capabilities without any message loss during the upgrade process. A test should also exist for a token change that fails due to an incompatible token, which should result in the connection entering the @FAILED@ state
+* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth#authorise@ is called or Ably requests a re-authentication (see "RTN22":#RTN22) and the state is @CONNECTED@, @CONNECTING@, or @DISCONNECTED@, the client will obtain a new token and send an @AUTH@ @ProtocolMessage@ to Ably. The @data@ attribute of the @ProtocolMessage@ must contain either a token string or an encoded @TokenDetails@ object with the @encoding@ attribute set accordingly
+* @(RTC8a)@ If the authentication token change is successful, then Ably will send a new @CONNECTED@ @ProtocolMessage@. The @connectionDetails@ provided in the @CONNECTED@ @ProtocolMessage@ must override any existing defaults, see "RTN21":#RTN21. A test should exist that performs an upgrade of capabilities without any loss of continuity or connectivity during the upgrade process. Another test should exist where the capabilities are downgraded resulting in Ably explicitly detaching the channel. That test must assert that the channel becomes detached soon after the token update and the reason is included in the channel state change event
```

Comment:
>> (does this spec item mean we may in the future be planning to expose capabilities in the connected message, btw?)

> No, and I don't think we should.

OK. I was about to ask what the point of even sending a CONNECTED is then if nothing's going to act on it and we're not exposing it (since none of the `connectionDetails` are things that could change with a new token), but I've just realised the answer to that is that realtime might want to change the `connectionKey`. So fair enough.